config/pmd/pmd.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0"?>
+<ruleset name="Custom Rules">
+    <description>
+        PMD configuration
+    </description>
+
+    <rule ref="category/java/bestpractices.xml">
+        <exclude name="MethodReturnsInternalArray"/>
+    </rule>
+
+    <rule ref="category/java/codestyle.xml">
+        <exclude name="CallSuperInConstructor"/>
+    </rule>
+
+    <rule ref="category/java/codestyle.xml/ShortClassName">
+        <properties>
+            <property name="minimum" value="4" />
+        </properties>
+    </rule>
+
+    <rule ref="category/java/design.xml">
+        <exclude name="LawOfDemeter"/>
+        <exclude name="UseUtilityClass"/>
+        <exclude name="LoosePackageCoupling"/>
+    </rule>
+
+    <rule ref="category/java/documentation.xml">
+        <exclude name="CommentRequired"/>
+        <exclude name="CommentSize"/>
+        <exclude name="UncommentedEmptyConstructor"/>
+    </rule>
+
+    <rule ref="category/java/errorprone.xml">
+        <exclude name="BeanMembersShouldSerialize"/>
+        <exclude name="MissingSerialVersionUID"/>
```

Comment:
It looks like a bug in your code

Reply:
I’ll read today about serialization and return to this conversation later :)